collect/collect.go

```diff
@@ -120,6 +112,18 @@ func (i *InMemCollector) Start() error {
 	return nil
 }
 
+func (i InMemCollector) createStartedCache(capacity int) *cache.DefaultInMemCache {
+	c := &cache.DefaultInMemCache{
+		Config: cache.CacheConfig{
+			CacheCapacity: capacity,
+		},
+		Metrics: i.Metrics,
+		Logger:  i.Logger,
+	}
+	c.Start()
```

Comment:
maybe we could just get rid of the start function? We don't have a matching stop function and it seems to just create some data structures that could be created upfront when we make the initial struct?